### PR TITLE
builder-next: fixes for rootless mode

### PR DESCRIPTION
**Pull request by tonistiigi**
@AkihiroSuda @tiborvass 

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Thanks, LGTM but compilation failing for Windows

**Comment by tonistiigi** [comment]
@AkihiroSuda updated

**Comment by AkihiroSuda** [comment]
```
08:30:29 daemon/info_unix.go:250:1:warning: exported method Daemon.Rootless should have comment or be unexported (golint)
08:30:30 Build step 'Execute shell' marked build as failure
```

File: builder/builder-next/controller.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func newController(rt http.RoundTripper, opt Opt) (*control.Controller, error) {
 		return nil, err
 	}
 
-	exec, err := newExecutor(root, opt.DefaultCgroupParent, opt.NetworkController)
+	exec, err := newExecutor(root, opt.DefaultCgroupParent, opt.NetworkController, opt.Rootless)
```
Comment (thaJeztah) [inline builder/builder-next/controller.go]: With more arguments being added, should we just pass `opt` here? (`newExecutor` is not exported, so ok to discuss for a follow-up)

File: builder/builder-next/executor_unix.go [hunk 2]
```diff
@@ -88,23 +91,26 @@ func (iface *lnInterface) init(c libnetwork.NetworkController, n libnetwork.Netw
 func (iface *lnInterface) Set(s *specs.Spec) {
 	<-iface.ready
 	if iface.err != nil {
+		logrus.Errorf("failed setting networking spec: %+v", iface.err)
```
Comment (thaJeztah) [inline builder/builder-next/executor_unix.go]: For consistency with other errors (easier to `grep`):

```suggestion
		logrus.Errorf("failed to set networking spec: %+v", iface.err)
```

or perhaps this (but I guess we'd loose the `%+v` verbosity)

```suggestion
		logrus.WithError(iface.err).Error("failed to set networking spec")
```

**Review by thaJeztah — APPROVED** [review]
LGTM, but left some suggestions, and the `failed setting` -> `failed to set` would be nice to have 😅

**Review by vdemeester — APPROVED** [review]
LGTM :man_dancing:

File: builder/builder-next/controller.go (hunk 1 above)
Comment (tonistiigi) [reply builder/builder-next/controller.go]: Let's leave this to follow-up

**Comment by cpuguy83** [comment]
```
11:58:12 ----------------------------------------------------------------------
11:58:12 FAIL: docker_cli_run_test.go:1791: DockerSuite.TestRunInteractiveWithRestartPolicy
11:58:12 
11:58:12 assertion failed: 
11:58:12 Command:  /usr/local/cli/docker run -i --name test-inter-restart --restart=always busybox sh
11:58:12 ExitCode: 0
11:58:12 Error:    <nil>
11:58:12 Stdout:   
11:58:12 Stderr:   
11:58:12 
11:58:12 Failures:
11:58:12 ExitCode was 0 expected 11
11:58:13 
11:58:13 ----------------------------------------------------------------------
```

Doesn't seem related, but does look new to me.

**Pull request merged**